src/hotspot/share/gc/shared/workgroup.hpp

```diff
@@ -330,21 +330,18 @@ class SubTasksDone: public CHeapObj<mtInternal> {
   // The calling thread asserts that it has attempted to claim all the tasks
   // that it will try to claim.  Tasks that are meant to be skipped must be
   // explicitly passed as extra arguments. Every thread in the parallel task
-  // must execute this.  (When the last thread does so, the task array is
-  // cleared.)
-  //
-  // n_threads - Number of threads executing the sub-tasks.
-  void all_tasks_completed(uint n_threads) {
-    all_tasks_completed_impl(n_threads, nullptr, 0);
+  // must execute this.
+  void all_tasks_completed() {
+    DEBUG_ONLY(all_tasks_completed_impl(nullptr, 0);)
   }
 
   // Augmented by variadic args, each for a skipped task.
   template<typename T0, typename... Ts,
           ENABLE_IF(Conjunction<std::is_same<T0, Ts>...>::value)>
-  void all_tasks_completed(uint n_threads, T0 first_skipped, Ts... more_skipped) {
+  void all_tasks_completed(T0 first_skipped, Ts... more_skipped) {
```

Comment:
I think this overload should be treated as the primary that the documentation applies to, with the no-arg overload following and being commented as being the base case for the variadic function.

Reply:
Done.